community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/DataFactories.java

```diff
@@ -427,4 +443,25 @@ else if ( isRecognizedType( typeSpec ) )
     {
         return Iterables.iterable( factories );
     }
+
+    public static Pair<String,String> splitTypeSpecAndOptionalParameter( String typeSpec )
+    {
+        String optionalParameter = null;
+        String newTypeSpec = typeSpec;
+        //TODO: This could be done with some clever regex magic
```

Comment:
Most likely :)

Reply:
`(?<newTypeSpec>.+?)(\{(?<optionalParameter>.*)\})?$`

I find https://regex101.com/ very helpful for this.